File: voom-fe/src/app/shared/rest/authentication/authentication.api.ts
import {Api} from '../api';
import {ApiClient} from '../api-client';
import {LoginDto, TokenDto, RegistrationDto, ResetPasswordDto, User, VerifyTokenDto} from './authentication.model';
import {RequestConfig} from '../rest.model';

export class AuthenticationApi extends Api {

  constructor(apiClient: ApiClient) {
    super(apiClient);
  }

  login(body: LoginDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
    }

    return this.apiClient.post<LoginDto, TokenDto>(`/api/auth/login`, body, config);
  }

  register(body: RegistrationDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
    }

    return this.apiClient.post<RegistrationDto, User>('/api/auth/register', body, config);
  }

  refreshToken(body: string) {
    const config: RequestConfig = {
      headers: {

        accept: 'application/json',
        contentType: 'application/json',
      }
    };

    return this.apiClient.post<string, TokenDto>(`/api/auth/refreshToken`, body, config);
  }

  verifyUser(token: string) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
    };

    return this.apiClient.post<VerifyTokenDto, void>(`/api/auth/verifyUser`, { token: token }, config);
  }


  forgotPassword(email: string) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
    };

    const body = { email: email };

    return this.apiClient.post<Object, void>(`/api/auth/forgotPassword`, body, config);
  }

  resetPassword(body: ResetPasswordDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
    };

    return this.apiClient.post<ResetPasswordDto, string>(`/api/auth/resetPassword`, body, config);
  }

}

File: voom-app/app/src/main/java/inc/visor/voom/app/shared/api/AuthenticationApi.java
package inc.visor.voom.app.shared.api;

import inc.visor.voom.app.shared.dto.authentication.ForgotPasswordDto;
import inc.visor.voom.app.shared.dto.authentication.LoginDto;
import inc.visor.voom.app.shared.dto.authentication.RegistrationDto;
import inc.visor.voom.app.shared.dto.authentication.ResetPasswordDto;
import inc.visor.voom.app.shared.dto.authentication.TokenDto;
import inc.visor.voom.app.shared.dto.authentication.UserDto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface AuthenticationApi {

    @POST("/api/auth/login")
    Call<TokenDto> login(@Body LoginDto dto);

    @POST("/api/auth/register")
    Call<UserDto> register(@Body RegistrationDto dto);

    @POST("/api/auth/refreshToken")
    Call<TokenDto> refreshToken(@Body String refreshToken);

    @POST("/api/auth/verifyUser")
    Call<Void> verifyUser(@Body String token);

    @POST("/api/auth/forgotPassword")
    Call<Void> forgotPassword(@Body ForgotPasswordDto dto);

    @POST("/api/auth/resetPassword")
    Call<Void> resetPassword(@Body ResetPasswordDto dto);


}
